Exiled.Events/Patches/Generic/PlayerHasHintPatch.cs

```diff
@@ -0,0 +1,69 @@
+// -----------------------------------------------------------------------
+// <copyright file="PlayerHasHintPatch.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+    using System.Collections.Generic;
+    using Exiled.API.Features;
+    using HarmonyLib;
+    using Hints;
+    using MEC;
+
+    /// <summary>
+    /// Patches <see cref="HintDisplay.Show(Hint)"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(HintDisplay), nameof(HintDisplay.Show))]
+    public static class PlayerHasHintPatch
+    {
+#pragma warning disable SA1313 // Parameter names should begin with lower-case letter
+#pragma warning disable SA1600 // Elements should be documented
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+        // Creating a list for coroutine check
+        private static List<Player> plyHaveCoroutine = new List<Player>();
+
+        public static void Postfix(HintDisplay __instance, Hint hint)
+        {
+            // Try to get the player, if it doesn't exist, just return
+            Player player = Player.Get(__instance.gameObject);
+            if (player == null)
+                return;
+
+            // Set HasHint to true, and then delays to false for the duration of the hint
+            player.HasHint = true;
+
+            // Set HintDisplayTime to the duration of the last hint
+            // This prevents to set HasHint to false before the hint has finished
+            player.HintRemainingTime = hint.DurationScalar;
+
+            // If not contains, run coroutine
+            if (!plyHaveCoroutine.Contains(player))
```

Comment:
Hints don't queue up like broadcasts, they overwrite eachother.
This check would fail when a plugin shows a hint for x seconds, then another one immediately after with a duration lower or higher than x.

So, if that list already contains a coroutine, you should kill it and run a new one.

Reply:
The check would not fail because I was using HintRemainingTime (my 🧠 went brrrr) and that's neither effective nor optimized, or i think so, but when you told me this I came up with another more effective idea.